src/shapes.ts

```diff
@@ -91,6 +91,16 @@ export interface OpenFinWindow {
     id: number;
     name: string;
     uuid: string;
+    preloadState?: { // set if window has a preload script
+        src: String;
+        state:
+            'idle'| // initial state, not emitted, really here for internal use
```

Comment:
@HarsimranSingh i'm okay with 'initial', the reason for 'idle': - script is in the queue/list and is 'idle', just sits there, maybe will be picked up later to download and execute maybe not, who knows? just being idle there...

@datamadic @rdepena @StevenEBarbaro @joneit  what are your thoughts on this/these?